src/RolloutScorer/RolloutScorerAzureFunction/RolloutScorerFunction.cs

```diff
@@ -1,135 +1,187 @@
+
+                        _logger.LogInformation($"INFO: Finding repo config for {currentRepo}...");
+                        RepoConfig currentRepoConfig = Configs.DefaultConfig.RepoConfigs[currentRepo];
+                        _logger.LogInformation($"INFO: Repo config: {currentRepoConfig.Repo}");
+
+                        _logger.LogInformation($"INFO: Finding AzDO config for {currentRepoConfig.AzdoInstance}...");
+                        AzdoInstanceConfig currentAzdoInstanceConfig = Configs.DefaultConfig.AzdoInstanceConfigs[currentRepoConfig.AzdoInstance];
+
                         RolloutScorer.RolloutScorer rolloutScorer = new RolloutScorer.RolloutScorer
                         {
                             Repo = deploymentGroup.Key,
-                            RolloutStartDate = deploymentGroup.First().Started.GetValueOrDefault().Date,
+                            RolloutStartDate = deploymentGroup.First().Started.GetValueOrDefault().Date,  // Is this going to blow up if GetValueOrDefault returns null?
                             RolloutWeightConfig = Configs.DefaultConfig.RolloutWeightConfig,
                             GithubConfig = Configs.DefaultConfig.GithubConfig,
-                            Log = log,
+                            Log = _logger,
+                            RepoConfig = currentRepoConfig, 
+                            AzdoConfig = currentAzdoInstanceConfig, 
+                            KeyVaultClient = _keyVaultClient
                         };
-                        log.LogInformation($"INFO: Finding repo config for {rolloutScorer.Repo}...");
-                        rolloutScorer.RepoConfig = Configs.DefaultConfig.RepoConfigs
-                            .Find(r => r.Repo == rolloutScorer.Repo);
-                        log.LogInformation($"INFO: Repo config: {rolloutScorer.RepoConfig.Repo}");
-                        log.LogInformation($"INFO: Finding AzDO config for {rolloutScorer.RepoConfig.AzdoInstance}...");
-                        rolloutScorer.AzdoConfig = Configs.DefaultConfig.AzdoInstanceConfigs
-                            .Find(a => a.Name == rolloutScorer.RepoConfig.AzdoInstance);
-
-                        log.LogInformation($"INFO: Fetching AzDO PAT from KeyVault...");
-                        using (KeyVaultClient kv = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(tokenProvider.KeyVaultTokenCallback)))
-                        {
-                            rolloutScorer.SetupHttpClient(
-                                (await kv.GetSecretAsync(rolloutScorer.AzdoConfig.KeyVaultUri, rolloutScorer.AzdoConfig.PatSecretName)).Value);
-                        }
+
+                        _logger.LogInformation($"INFO: Fetching AzDO PAT from KeyVault...");
+                        
+                        rolloutScorer.SetupHttpClient(
```

Comment:
Ideally, SetupHttpClient and SetupGithubClient should be ran internally in the RolloutScorer object when it's created or initialized.